File: Isomorphic Strings/Isomorphic_Strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character,Character> mpp = new HashMap<>();
        HashMap<Character,Character> mpp2 = new HashMap<>();
        int n = s.length();
        for(int i =0; i<n; i++) {
            char si = s.charAt(i);
            char ti = t.charAt(i);
            if(mpp.containsKey(si)) {
                if(mpp.get(si) != ti) return false;
            }
            if(mpp2.containsKey(ti)) {
                if(mpp2.get(ti) != si) return false;
            }
            mpp.put(si,ti);
            mpp2.put(ti,si);
        }
        return true;
    }
}

File: Isomorphic Strings/Isomorphic_Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        mpp = {}
        mpp2 = {}
        for si,ti in zip(s,t):
            if si in mpp:
                if mpp[si] != ti:
                    return False
            if ti in mpp2:
                if mpp2[ti] != si:
                    return False
            mpp[si] = ti
            mpp2[ti] = si
        return True
        